packages/drawer/src/test/mwc-drawer.test.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// taze: chai from //third_party/javascript/chai:closure_chai
+// taze: sinon from //third_party/javascript/sinon:sinon_lib
+
+import {Drawer} from '@material/mwc-drawer';
+
+const SCRIM_SELECTOR = '.mdc-drawer-scrim';
+const HEADER_SELECTOR = '.mdc-drawer__header';
+const DISMISSIBLE_CLASS = 'mdc-drawer--dismissible';
+
+suite('mwc-drawer', () => {
+  let element: Drawer;
+
+  setup(async () => {
+    element = document.createElement('mwc-drawer');
```

Comment:
Button tests are pretty old and not too clear when it comes to templating. Do you mind using the patterns in the dialog and list tests using `fixture` and `TemplateResult`s?

example:
* [templating](https://github.com/material-components/material-components-web-components/blob/master/packages/dialog/src/test/mwc-dialog.test.ts#L89)
* [usage](https://github.com/material-components/material-components-web-components/blob/master/packages/dialog/src/test/mwc-dialog.test.ts#L254) (perhaps you want this in setup rather than test though)
* [fixture implementation](https://github.com/material-components/material-components-web-components/blob/master/test/src/util/helpers.ts#L91)

Reply:
I have another method of using these I was experimenting on in list if you want to take a look. let me know if you think that is too complex or ugly. Looking for a bit of feedback for testing standardization:

* [generator](https://github.com/material-components/material-components-web-components/blob/master/packages/list/src/test/mwc-list.test.ts#L60)
* [usage](https://github.com/material-components/material-components-web-components/blob/master/packages/list/src/test/mwc-list.test.ts#L243)

Reply:
thats a cool idea, the generator. i do think there's benefit in less abstraction though, too, as you can clearly see the properties etc being set per test fixture before. 

its a decision between simplicity of writing tests (you just have one generator, many fixture props) and reading tests (you have many fixtures, with clear individual 'hard-coded' properties).

Reply:
to be fair though you can see the non-default stuff being set in each individual test, so as long as those properties are well-named, its probably just as readable as the fixtures before.

Reply:
i just tried it on the button tests. what i would say is...

it quickly gets repetitive, adding a fixture in each test. if the particular property or method you're testing only needs one test, you don't need a setup function, so you end up with two lines per test (fixt and element being set). which is a lot of overhead compared to, at the top of each test, just setting the individual property.

it feels like we're hiding the defaults from tests. if our 'generator' _always_ sets its own defaults, we're not really testing what they actually are behind the scenes.

Reply:
Though, I think the generator is useful for cases such as list-item that has quite a lot of properties. Perhaps less-useful in components like button.

* Very good point. Though I'd argue you'd also have to do this for all variants as well as this is important for elements such as `textfield` with its `outlined` variant. I think your solution is a great first-step to standardizing this.
* I like this. Let's do this
* Sure, this works